srai/regionizers/h3_regionizer.py

```diff
@@ -0,0 +1,191 @@
+"""
+H3 regionizer.
+
+This module exposes Uber's H3 Hexagonal Hierarchical Geospatial Indexing System [1] as a regionizer.
+
+Note:
+    The default API [2] was chosen (basic_str) to ease the implementation.
+    It may be beneficial to try the NumPy API for computationally-heavy work.
+
+References:
+    [1] https://uber.github.io/h3-py/
+    [2] https://uber.github.io/h3-py/api_comparison
+
+"""
+
+from typing import List
+
+import geopandas as gpd
+import h3
+import pandas as pd
+from functional import seq
+from shapely import geometry
+
+
+class H3Regionizer:
+    """
+    H3 Regionizer.
+
+    H3 Regionizer allows the given geometries to be divided
+    into H3 cells - hexagons with pentagons as a very rare exception
+
+    """
+
+    def __init__(self, resolution: int, buffer: bool = True) -> None:
+        """
+        Init H3Regionizer.
+
+        Args:
+            resolution (int): Size of the cells.
```

Comment:
I think we can stick with the word `Resolution` instead of size here, even if it doubles. Size is a complete inversion in this context (higher resolution == smaller size).
```suggestion
            resolution (int): Resolution of the cells. See [1] for a full comparison.
```

And then attach this https://h3geo.org/docs/core-library/restable/ as a reference.